packages/rest/src/server.ts

```diff
@@ -31,6 +36,30 @@ export const setupServer = async (agent: Agent, config: ServerConfig) => {
     proofEvents(agent, config)
   }
 
+  socketServer.on('connection', (stream) => {
+    webSocketEvents(socketServer, stream, agent, config)
+  })
+
+  // Create all transports
+  const httpInboundTransport = new HttpInboundTransport({ app: server, port: config.port })
+  const httpOutboundTransport = new HttpOutboundTransport()
+  const wsInboundTransport = new WsInboundTransport({ server: socketServer })
+  const wsOutboundTransport = new WsOutboundTransport()
+
+  // Register all Transports
+  agent.registerInboundTransport(httpInboundTransport)
+  agent.registerOutboundTransport(httpOutboundTransport)
+  agent.registerInboundTransport(wsInboundTransport)
+  agent.registerOutboundTransport(wsOutboundTransport)
+
+  await agent.initialize()
+
+  httpInboundTransport.server?.on('upgrade', (request, socket, head) => {
+    socketServer.handleUpgrade(request, socket as Socket, head, (socket) => {
+      socketServer.emit('connection', socket, request)
+    })
```

Comment:
The rest package adds a rest api around the agent, but doesn't configure the agent for you. This allows to use your own agent with the rest package.

I think we should reuse the server that we are using for the rest api (so the `server` variable) for the weboscket server.

```
  socketServer.on('connection', (stream) => {
    webSocketEvents(socketServer, stream, agent, config)
  })

  server.on('upgrade', (request, socket, head) => {
    socketServer.handleUpgrade(request, socket as Socket, head, (socket) => {
      socketServer.emit('connection', socket, request)
    })
```